### PR TITLE
Focal drop condition environment from hotplug

### PR DESCRIPTION
**Pull request by blackboxsw**
- [x] I have signed the CLA: https://ubuntu.com/legal/contributors
- [x] I have added my Github username to ``tools/.github-cla-signers``
- [x] I have included a comprehensive commit message using the guide below
- [ ] I have added unit tests to cover the new behavior under ``tests/unittests/``
  - Test files should map to source files i.e. a source file ``cloudinit/example.py`` should be tested by ``tests/unittests/test_example.py``
  - Run unit tests with ``tox -e py3``
- [x] I have kept the change small, avoiding unnecessary whitespace or non-functional changes.
- [x] I have added a reference to issues that this PR relates to in the PR message (Refs #1234, Fixes: GH-1234)
- [ ] I have updated the documentation with the changed behavior.
  - If the change doesn't change the user interface and is trivial, this step may be skipped.
  - Cloud-config documentation is generated from the jsonschema.
  - Generate docs with ``tox -e docs``.

## Proposed Commit Message
```
fix(systemd): drop unsupported ConditionEnvironment from hotplug

Additionally drop ConditionEnvironment config from hotplug units
as it is not supported on Focal's version of systemd.

Fixes: GH-5402
```

## Test Steps
```
quilt push -a
git diff debian/changes/*hotplug*
quilt pop -a
```

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Review by aciba90 — COMMENTED** [review]
Thanks, @blackboxsw, for addressing this.
```
quilt push -a
git diff systemd/*hotplug*
quilt pop -a
```

The patch changes and the above commands look good, but [CI](https://github.com/canonical/cloud-init/actions/runs/9667347690/job/26668956668?pr=5448) and I are having troubles building the package:

```
$ build-package 
Preparing worktree (detached HEAD fbdfb2e71)
HEAD is now at fbdfb2e71 fix(systemd): drop unsupported ConditionEnvironment from hotplug
Found UNRELEASED in debian/changelog. Do you really want to build-package ? (y/n)  y
pkg_name=cloud-init pkg_ver=24.1.3-0ubuntu1~20.04.4 upstream_ver=24.1.3 suite=UNRELEASED
Check for an upstream cloud-init_24.1.3.orig.tar.gz on which to base this debdiff ? (y/n)  n
========================================
./tools/make-tarball --output=../dl/cloud-init_24.1.3.orig.tar.gz
../dl/cloud-init_24.1.3.orig.tar.gz
========================================
Removing upstream's debian dir in favor of packaging.
building with: -d -S -nc
 dpkg-buildpackage -d -us -uc -ui -S -nc
dpkg-buildpackage: info: source package cloud-init
dpkg-buildpackage: info: source version 24.1.3-0ubuntu1~20.04.4
dpkg-buildpackage: info: source distribution UNRELEASED
dpkg-buildpackage: info: source changed by Chad Smith <chad.smith@canonical.com>
 dpkg-source --before-build .
dpkg-source: info: using patch list from debian/patches/series
dpkg-source: info: applying retain-apt-partner-pocket.patch
dpkg-source: info: applying expire-on-hashed-users.patch
dpkg-source: info: applying retain-netplan-world-readable.patch
dpkg-source: info: applying netplan99-cannot-use-default.patch
dpkg-source: info: applying retain-old-groups.patch
dpkg-source: info: applying keep-dhclient-as-priority-client.patch
dpkg-source: info: applying revert-551f560d-cloud-config-after-snap-seeding.patch
dpkg-source: info: applying status-do-not-remove-duplicated-data.patch
dpkg-source: info: applying retain-apt-pre-deb822.patch
dpkg-source: info: applying status-retain-recoverable-error-exit-code.patch
dpkg-source: info: applying retain-ec2-default-net-update-events.patch
dpkg-source: info: applying cli-retain-file-argument-as-main-cmd-arg.patch
dpkg-source: info: applying drop-unsupported-systemd-condition-environment.patch
patching file systemd/cloud-config.service.tmpl
Hunk #1 FAILED at 6.
1 out of 1 hunk FAILED
patching file systemd/cloud-final.service.tmpl
patching file systemd/cloud-init-local.service.tmpl
patching file systemd/cloud-init.service.tmpl
patching file systemd/cloud-init.target
patching file systemd/cloud-init-hotplugd.service
patching file systemd/cloud-init-hotplugd.socket
dpkg-source: info: the patch has fuzz which is not allowed, or is malformed
dpkg-source: info: if patch 'drop-unsupported-systemd-condition-environment.patch' is correctly applied by quilt, use 'quilt refresh' to update it
dpkg-source: info: if the file is present in the unpacked source, make sure it is also present in the orig tarball
dpkg-source: info: restoring quilt backup files for drop-unsupported-systemd-condition-environment.patch
dpkg-source: error: LC_ALL=C patch -t -F 0 -N -p1 -u -V never -E -b -B .pc/drop-unsupported-systemd-condition-environment.patch/ --reject-file=- < wtd/debian/patches/drop-unsupported-systemd-condition-environment.patch subprocess returned exit status 1
dpkg-buildpackage: error: dpkg-source --before-build . subprocess returned exit status 2
debuild: fatal error at line 1184:
dpkg-buildpackage -d -us -uc -ui -S -nc failed
FAILED: debuild -d -S -nc
```

The same happens running `sbuild -d focal` or `debuild -uc -us`. 
Not sure what happens. Do we need to refresh the patches, or is this something related to how we deal with the upstream tarball?

**Comment by blackboxsw** [comment]
@aciba90 I should have probably performed a new_upstream_snapshot.py on this branch after adapting the quilt patch to ensure the ubuntu/focal branch remained buildable without requiring one to merge into upstream/main first (which would have resolved things for daily recipe builds but not the local branch as you saw).

I have since refactored the original commit https://github.com/canonical/cloud-init/pull/5448/commits/7a2163c281511d7a03e0089931520fb92bcaa9ad, sync'd any unreleased changes from [ubuntu/focal-24.1.x:debian/changelog](https://github.com/canonical/cloud-init/blob/ubuntu/focal-24.1.x/debian/changelog#L1-L22) and  performed `new_upstream_snapshot.py -c 24.3` to actually plan on releasing this branch.


Please feel free to test this package build fully:
````
cd /tmp
mkdir ../dl
cd dl
wget https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/cloud-init/24.2-0ubuntu1/cloud-init_24.2.orig.tar.gz
cd ../
git clone git@github.com:canonical/cloud-init.git -o upstream
cd cloud-init
git remote add blackboxsw git@github.com:blackboxsw/cloud-init.git 
git checkout blackboxsw/ubuntu/focal -B ubuntu/focal
quilt push -a
tox .-e py3
quilt pop -a
build-package
sbuild --dist=focal --arch=amd64 --arch-all ../out/cloud-init_24.2*.dsc
```

**Review by aciba90 — APPROVED** [review]
LGTM, thanks!

- [x] Version looks good after new upstream snapshot
- [x] Patches apply cleanly
- [x] sbuild
- [x] quilt push -a && tox -e py3 && quitl pop -a

File: debian/changelog [hunk 1]
```diff
@@ -27,9 +27,14 @@ cloud-init (24.1.3-0ubuntu1~20.04.4) UNRELEASED; urgency=medium
     - d/p/status-do-not-remove-duplicated-data.patch
     - d/p/status-retain-recoverable-error-exit-code.patch
     - d/p/revert-551f560d-cloud-config-after-snap-seeding.patch
-  * Upstream snapshot based on upstream/main at debafbc9.
+  * refresh patches:
```
Comment (TheRealFalcon) [inline debian/changelog:30]: I don't think this is a blocker, but we have a double `refresh patches` section here.

**Pull request merged**